### PR TITLE
Raise fatal error when inspector_modules.js cannot be loaded.

### PR DESCRIPTION
**Pull request by jeduden**
When the debugger is connected and the module cannot be loaded the debugger does not work. Without reporting the error is can get very difficult to troubleshoot the issue.

### CODE REVIEW

**Comment by KristinaKoeva** [comment]
Hey @jeduden, thank you very much for your contribution !
Would you please explain what are the cases when the `inspector_modules` fails to load ? Do you have any custom modifications in it ?

**Comment by jeduden** [comment]
@KristinaKoeva  this patch is a by-product of my research related to issue #717 . 

TLDR:
When having a non standard js delivery system and/or when one is not conscious of the fact that the debugger needs a couple of js files from the tns_modules in order to work. It is quiet a journey to troubleshoot the issue, since i had to understand the cmake build files and the internals of how the debugger and the runtime work hand in hand in order to diagnose the issue.

With this PR i hope to short-circuit the journey and to report directly back that the debugger is not functional if this error occurs.

Full version: 
The ios nativescript runtime is integrated in an existing application, which has it's own systems to download js.
I was looking into why the js debugger didn't work with this application. The first step was to simplify the scenario and have only one runtime but still the custom js downloader, in that simplified scenario debugging also didn't work. 

The backend could connect and i was able to see source code but breakpoints and so on didn't work. There wasn't any feedback in the logs why the debugger didn't work.

I  build a debuggable version of the ios runtime to see what exactly failed. That step was quiet tricky, since the build scripts seem to support only release builds out of the box.

After i was able to step through the native code, i was able to see that the debugger code tried to load inspector_modules.js which i didn't expect.  But stepping over the load call, i didn't see any  obvious issues. Only later i realized that the call actually returns a promise which needs interpretation before figuring out if load actually worked.

And when i add in the error handler i was finally able to diagnose that the inspector modules were not found. And when I fixed delivery of the tns_modules to app folder i was able to confirm that the inspector modules are loading and debugger started to work as well.

**Comment by KristinaKoeva** [comment]
The thing that isn't clear to me is why the debugger does not work. I've created a simple {N} project and tried the following scenarios:
1. Added `throw "Error"` on the first line of the `inspector_modules` file
1. Made a syntax error in the `inspector_modules` file
1. Deleted the `inspector_modules` file

Neither of them result in a broken debugger experience. 
I completely agree that we should provide logging but since this file is only used for enhancing the debugger experience I don't think that we should kill the user application when something is wrong.

Would you please share something simple that demonstrates the problem ?

**Comment by jeduden** [comment]
@KristinaKoeva  
After reading your email, I was looking into reproducing the issue with a minimal project.

1. In this tag the debugger works: https://github.com/jeduden/min-app-inspector-not-working/releases/tag/debugger-works-out-the-box

Confirming that the inspector modules are not required. So I continued looking why it didn't work for me.

2. In this tag the debugger doesnt work: https://github.com/jeduden/min-app-inspector-not-working/releases/tag/debugger-doesnt-workx
    it uses JSEvaluateString to execute a script. not executeModule

3. In this tag the debugger works again: https://github.com/jeduden/min-app-inspector-not-working/releases/tag/debugger-works-with-tns_modules-again
    it includes the tns_modules in the folder. and the debugger works again.

4. in this tag https://github.com/jeduden/min-app-inspector-not-working/releases/tag/debugger-works-after-calling-exec-module-once
    i tested the hypothesis that the debugger starts working after the first call to executeModule.
   which it does.

I guess loadAndEvaluateModule has a side-effect that initialises/fixes some state essential for the debugger to work, but I am not sure yet why.

**Comment by KristinaKoeva** [comment]
Hey @jeduden,
I had some troubles running the project that you've attached under the second point - https://github.com/jeduden/min-app-inspector-not-working/releases/tag/debugger-doesnt-work
So I've created a new plain {N} project, I've deleted the contents of the app folder and I've replaced the content of main.m in the new project with the content of main.m from the project that you've attached. Still everything works on my side, the debugger waits at the `debugger;` statement and I am able to attach without problems. 

If you don't mind I will close this PR and we will revisit it once we decide whether we should terminate the application or log a warning when there are errors in the `inspector_modules.js` file. You can open an issue where we could try to find what's causing your debugger problems. Is that OK for you ?

**Comment by jeduden** [comment]
@KristinaKoeva sure. no worries, reproducing this is tricky. indeed. 

One question regarding inspector_modules.js if the module is "optional" why is it loaded in the first place ? What functionality does it provide ?

**Comment by KristinaKoeva** [comment]
Some time ago we've implemented [a way to extend the debugger through JavaScript code](https://github.com/NativeScript/ios-runtime/pull/375). The main reason for this was that the runtime is not aware of the UI aspects of the application hence the debugger cannot provide CSS or DOM debugging. The CSS and DOM information is part of the {N} modules that are all written in JavaScript. So we've decided to introduce the `inspector_modules.js` file that should be loaded before everything else in the application. In this file the {N} modules are registering specific handlers that would eventually extend the debugger with CSS or DOM debugging. Currently this file contains only an extension for debugging Network request, but we plan to add more extensions in the future. Here is the code of the [`inspector_modules.js` file](https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/inspector_modules.ios.ts) This of course would not be available if you don't use {N} modules in your application.
